COMPATIBILITY.md

```diff
@@ -0,0 +1,102 @@
+# `rules_rust` backwards compatibility policy
+
+This document defines the backwards compatibility policy for rules\_rust and
+defines the process for making compatibility breaking changes. Any exception to
+this process will have to be thoroughly discussed and there will have to be a
+consensus between maintainers that the exception should be granted.
+
+## What is a compatibility breaking change?
+
+A backwards compatible change has the property that a build that is green,
+correct, is using stable APIs of `rules_rust,` and is using a supported version
+of Bazel before the change is still green and correct after the change. In
+practice it means that users should not have to modify their project source
+files, their `BUILD` files, their `bzl` files, their `rust_toolchain`
+definitions, their platform definitions, or their build settings.
+
+A backwards compatible release has the property that it only contains backward
+compatible changes.
+
+`rules_rust` follow [SemVer 2.0.0](https://semver.org/). `rules_rust` promise
+that all minor version number and patch version number releases only contain
+backwards compatible changes.
+
+`rules_rust` don't make any guarantees about compatibility between a released
+version and the state of the `rules_rust` at HEAD.
+
+Backwards incompatible changes will have to follow a process in order to be
+released. Users will be given at least one release where the old behavior is
+still present but deprecated to allow smooth migration of their project to the
+new behavior.
+
+## What Bazel versions are supported?
+
+`rules_rust` support the current Bazel LTS at the time of a `rules_rust`
+release.
+
+Support for the current Bazel rolling release is on the best effort basis. If
+the CI build of `rules_rust` against the current Bazel rolling release is green,
+it has to stay green. If the build is already red (because the new Bazel rolling
+release had incompatible changes that broke `rules_rust`), it is acceptable to
+merge PRs leaving the build red as long as the reason for failure remains the
+same. We hope red CI with the current rolling release will be rare.
+
+`rules_rust` don't promise all new `rules_rust` features available to the
+current Bazel rolling release to be be available to the current Bazel LTS
+release (because Bazel compatibility policy doesn't allow us to make that
+promise, and some new features of `rules_rust` require new Bazel features that
+are only available in Bazel rolling releases). `rules_rust` do promise that new
+features available to the current Bazel rolling release will be available to the
+next Bazel LTS release at latest.
+
+Whenever there is a new Bazel LTS release, all releases of `rules_rust` in the
+first 3 months will support both Bazel LTS versions.
+
+## What host platforms are supported?
+
+The only platform subject to backwards compatibility policy is `Linux` on
```

Comment:
I would like to extend this to Darwin x86_64 - is there a reason we don't think this is practical?

Reply:
Related, why don't we support everything Bazel supports? It'd also probably be good to indicate this support is for execution environments.

Reply:
Let's move this to a separate issue? (In short I have no objections but at least from the start I'd like to have a champion with available capacity to fix things (for example enabling all tests on mac and windows that are currently disabled because reasons). Or making sure all features work on mac and windows just like they do on linux (can we use ld.exe to link?, can we pass env vars to test executions? that kind of stuff)

Reply:
I created separate issues for this, let's continue there.